### PR TITLE
[core] fix(NumericInput): don't apply textbox role

### PR DESCRIPTION
**Pull request by bvandrc**
#### Checklist

- [N/A] Includes tests
- [N/A] Update documentation

#### Changes proposed in this pull request:

update the `type` prop for the `<input>` element created in a `numericInput`. `role` prop no longer needed due to given `type`

### CODE REVIEW

File: packages/core/src/components/forms/numericInput.tsx [hunk 1]
```diff
@@ -421,19 +421,15 @@ export class NumericInput extends AbstractPureComponent2<HTMLInputProps & Numeri
...
                 asyncControl={this.props.asyncControl}
                 autoComplete="off"
                 id={this.numericInputId}
-                role={hasSpinButtons ? "spinbutton" : "textbox"}
+                role={this.props.allowNumericCharactersOnly ? "spinbutton" : "textbox"}
+                type={this.props.allowNumericCharactersOnly ? "number" : "text"}
                 {...inputGroupHtmlProps}
-                aria-valuemax={this.props.max}
-                aria-valuemin={this.props.min}
-                aria-valuenow={valueAsNumber}
```
Comment (bvandrc) [inline packages/core/src/components/forms/numericInput.tsx]: Realized that these are probably not needed due to the fact that the `<input>` already has `min` and `max` props, and a value.

File: packages/core/src/components/forms/numericInput.tsx [hunk 2]
```diff
@@ -421,19 +421,15 @@ export class NumericInput extends AbstractPureComponent2<HTMLInputProps & Numeri
...
-        const valueAsNumber = this.getCurrentValueAsNumber();
-        const hasSpinButtons = this.props.buttonPosition !== undefined && this.props.buttonPosition !== "none";
 
         return (
             <InputGroup
                 asyncControl={this.props.asyncControl}
                 autoComplete="off"
                 id={this.numericInputId}
-                role={hasSpinButtons ? "spinbutton" : "textbox"}
+                role={this.props.allowNumericCharactersOnly ? "spinbutton" : "textbox"}
```
Comment (nerdstep) [inline packages/core/src/components/forms/numericInput.tsx]: The `role` would be implicit here based on setting the `type` correctly

Comment (bvandrc) [reply packages/core/src/components/forms/numericInput.tsx]: got it, removed the role prop

File: packages/core/src/components/forms/numericInput.tsx (hunk 1 above)
Comment (adidahiya) [reply packages/core/src/components/forms/numericInput.tsx]: Is there any documentation for this? Would it hurt to leave them in?

Comment (bvandrc) [reply packages/core/src/components/forms/numericInput.tsx]: I don't think it hurts, just unneccessary

Comment (adidahiya) [reply packages/core/src/components/forms/numericInput.tsx]: Does the browser read min/max attributes when the input type is "text"? I feel like they could potentially be useful there... but again, I don't have specific documentation to support this idea.

Comment (bvandrc) [reply packages/core/src/components/forms/numericInput.tsx]: that is a very good point actually. We should probably keep these since the type is "text".

**Review by adidahiya — APPROVED** [review]

**Pull request merged**